x-pack/plugins/apm/public/components/app/service_overview/service_overview_transactions_table/index.tsx

```diff
@@ -124,140 +79,97 @@ export function ServiceOverviewTransactionsTable(props: Props) {
-        {
-          defaultMessage: 'Name',
+          return { [requestId]: metrics };
         }
-      ),
-      render: (_, { name, transactionType: type }) => {
-        return (
-          <TruncateWithTooltip
-            text={name}
-            content={
-              <TransactionDetailLink
-                serviceName={serviceName}
-                transactionName={name}
-                transactionType={type}
-                latencyAggregationType={latencyAggregationType}
-              >
-                {name}
-              </TransactionDetailLink>
-            }
-          />
-        );
-      },
-    },
-    {
-      field: 'latency',
-      name: getLatencyAggregationTypeLabel(latencyAggregationType),
-      width: px(unit * 10),
-      render: (_, { latency }) => {
-        return (
-          <SparkPlot
-            color="euiColorVis1"
-            compact
-            series={latency.timeseries ?? undefined}
-            valueLabel={asMillisecondDuration(latency.value)}
-          />
-        );
-      },
+      }
+      return fetchAggResults();
     },
-    {
-      field: 'throughput',
-      name: i18n.translate(
-        'xpack.apm.serviceOverview.transactionsTableColumnThroughput',
-        { defaultMessage: 'Throughput' }
-      ),
-      width: px(unit * 10),
-      render: (_, { throughput }) => {
-        return (
-          <SparkPlot
-            color="euiColorVis0"
-            compact
-            series={throughput.timeseries ?? undefined}
-            valueLabel={asTransactionRate(throughput.value)}
-          />
-        );
-      },
-    },
-    {
-      field: 'errorRate',
-      name: i18n.translate(
-        'xpack.apm.serviceOverview.transactionsTableColumnErrorRate',
-        {
-          defaultMessage: 'Error rate',
-        }
-      ),
-      width: px(unit * 8),
-      render: (_, { errorRate }) => {
-        return (
-          <SparkPlot
-            color="euiColorVis7"
-            compact
-            series={errorRate.timeseries ?? undefined}
-            valueLabel={asPercent(errorRate.value, 1)}
-          />
-        );
-      },
-    },
-    {
-      field: 'impact',
-      name: i18n.translate(
-        'xpack.apm.serviceOverview.transactionsTableColumnImpact',
-        {
-          defaultMessage: 'Impact',
-        }
-      ),
-      width: px(unit * 5),
-      render: (_, { impact }) => {
-        return <ImpactBar value={impact ?? 0} size="m" />;
-      },
+    // only fetches metrics when requestId changes or transaction names changes
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [requestId, transactionNames]
+  );
+
+  const columns = getColumns({
+    serviceName,
+    latencyAggregationType,
+    transactionGroupsAggResults: transactionGroupsAggResults
+      ? transactionGroupsAggResults[requestId]
+      : undefined,
+  });
+
+  const isLoading =
+    status === FETCH_STATUS.LOADING ||
+    transactionGroupsAggResultsStatus === FETCH_STATUS.LOADING;
```

Comment:
should this not include NOT_INITIALIZED?

Reply:
If we include `NOT_INITIALIZED` and navigate to a future date range, the required fields for these API calls to happen won't be defined, causing the table to never stop showing the loading spinner.

Reply:
We'd only have that check for `transactionGroupsAggResultsStatus`. Would that still be the case then? Maybe when transaction groups is empty 🤔 But this doesn't this shortly hide and then show the loading spinner again, for maybe one frame? If it's not visible to the naked eye that's fine too.

Reply:
> Would that still be the case then?

Yes, it is the same case.

> But this doesn't this shortly hide and then show the loading spinner again, for maybe one frame?

Yes, it does hide and show the spinner, but it is practically impossible to notice it.